### PR TITLE
Add authors for release 5.1.0

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**

Adds new / existing contributors to DOI authors table. This is required for generating a DOI for Mantid 5.1.0

Could people request changes if their name is incorrect at all, or they would prefer a different form (e.g. Bob -> Robert) before 12:00 GMT tomorrow (30th Sept.). If I don't hear anything I'll assume all is correct.

----

I've @ people who did not have their name in the table previously.

@tolu28-coder 
@joseph-torsney 
@YannickMeinerzhagen 
@durong24 
@Matt-Cumber 
@clayton-tom 
@KedoKudo 
@gui-co 
@darominski 

---

To Test:

- Check there aren't any obvious typos in peoples names

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: tools/DOI/authors.py [hunk 1]
```diff
@@ -239,7 +240,22 @@
     'William F Godoy'         : 'Godoy, William F',
     'Islam, Fahima F'         : 'Islam, Fahima',
     'Mathieu Tillet'          : 'Tillet, Mathieu',
-    'StephenSmith'            : 'Smith, Stephen'
+    'MathieuTillet'           : 'Tillet, Mathieu',
+    'StephenSmith'            : 'Smith, Stephen',
```
Comment (AnthonyLim23) [inline tools/DOI/authors.py:244]: is already above https://github.com/mantidproject/mantid/pull/29644/files#diff-29d12c5945d69452fc4830199bec6ff4L217

Comment (DavidFair) [reply tools/DOI/authors.py:244]: This is just a side effect of the fact I had to add `MathieuTillet` with the other alias they used 
I haven't touched Stephen's mapping in this PR

**Review by martyngigg — APPROVED** [review]

**Comment by martyngigg** [comment]
Merging so we can create the DOI.

**Comment by DavidFair** [comment]
Just an FYI: I've generated our DOI for 5.1.0 at : http://dx.doi.org/10.5286/Software/Mantid5.1

**Comment by martyngigg** [comment]
Thanks @DavidFair

**Pull request merged**